### PR TITLE
Implement airy functions (except the `ai_zeros` and `bi_zeros` functions)

### LINKED ISSUE #750: Implement airy functions
Implement airy functions including `airy`, `airye`, `ai_zeros`, `bi_zeros`, `itairy`.
list in https://docs.scipy.org/doc/scipy/reference/special.html#airy-functions.

### PR DESCRIPTION
**Pull request by shantam-8**
## What do these changes do?

 The PR is implements all airy functions except for ai_zeros and bi_zeros.
## Related issue number

Attempts to fixes #750

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Comment by RandomY-2** [comment]
Hi, to implement airy functions, you can use the ```TensorTupleOp``` class I implemented in ```special.core.py```. Specifically, you just need to inherit this class and specify how many outputs the function will have. I also added an example with ```TensorAiry```, which you can use as a reference. For testing, you can just add the function name and tensor class to the corresponding test function for tuple returns (```test_unary_tuple_execution``` and ```test_unary_tuple_operand``` for example). I think the currently test function needs to be modified to test airy functions. For example, ```test_unary_tuple_operand``` only supports out tuple with two elements, but ```airy``` has four. So maybe you can pass in another parameter to specify the out tuple's size. If not I can modify it later. Thanks.

File: mars/tensor/special/airy.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+class TensorAiry(TensorTupleOp):
+    _func_name = "airy"
+    _n_outputs = 4
+
+
+@implement_scipy(spspecial.airy)
+@infer_dtype(spspecial.airy, multi_outputs=True)
+def airy(z, out=None, **kwargs):
+    op = TensorAiry(**kwargs)
+    return op(z, out=out)
```
Comment (RandomY-2) [inline mars/tensor/special/airy.py:32]: ```airy``` example

File: mars/tensor/special/tests/test_special_execution.py [hunk 2]
```diff
@@ -311,6 +315,7 @@ def test_quintuple_execution(setup, func):
         "fresnel",
         "modfresnelp",
         "modfresnelm",
+        "airy",
```
Comment (RandomY-2) [inline mars/tensor/special/tests/test_special_execution.py:318]: add the function name to the corresponding test case

**Comment by shantam-8** [comment]
Thanks a lot for your help @RandomY-2! I have implemented the `airy`, `airye`, and `itairy` functions and their tests are passing. However, I could not implement the `ai_zeros` and `bi_zeros` functions as they both require a positive scalar integer.

File: mars/tensor/special/airy.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import scipy.special as spspecial
+
+from ..utils import infer_dtype, implement_scipy
+from .core import TensorTupleOp, _register_special_op
+
+
+@_register_special_op
+class TensorAiry(TensorTupleOp):
+    _func_name = "airy"
+    _n_outputs = 2
```
Comment (shantam-8) [inline mars/tensor/special/airy.py:24]: Actual value of `n_ouptuts` = 4, but this is done to pass the `test_unary_tuple_operand` test in `test_special.py`. The same is done for `airye` and `itairy` functions.

Comment (RandomY-2) [reply mars/tensor/special/airy.py]: I updated the test function so the out tuple's size can be adjusted based on the operand. So using 4 for n_outputs should be fine now.

**Review by wjsi — APPROVED** [review]
LGTM

**Review by zhongchun — APPROVED** [review]
LGTM.

**Pull request merged**